pkg/build/api/v1/types.go

```diff
@@ -477,8 +478,23 @@ type SourceBuildStrategy struct {
 	// incremental flag forces the Source build to do incremental builds if true.
 	Incremental bool `json:"incremental,omitempty"`
 
-	// forcePull describes if the builder should pull the images from registry prior to building.
+	// forcePull describes if the builder should pull the images from registry
+	// prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// runtimeImage is an optional image that is used to run an application
+	// without unneeded dependencies installed.
+	RuntimeImage *kapi.ObjectReference `json:"runtimeImage,omitempty"`
+
+	// runtimeArtifacts holds the mapping between the artifacts paths in the
+	// builder image and the paths in the runtime image where they should be
+	// copied.
+	RuntimeArtifacts []ImageSourcePath `json:"runtimeArtifacts,omitempty"`
+
+	// runtimePullSecret is the name of a Secret that would be used for setting up
+	// the authentication for pulling the Docker runtime images from private
+	// Docker registries.
+	RuntimePullSecret *kapi.LocalObjectReference `json:"runtimePullSecret,omitempty"`
```

Comment:
the discussion definitely wasn't conclusive, but i'm going to propose we go with the single field.  note that as @liggitt pointed out, this means we need to be a little more sophisticated in the logic that picks the secret in the first place, since we need to find a secret that matches both image pullspecs.